### ISSUE TITLE
Error in BPR loss function

### ISSUE THREAD

**Issue by PeterZaidel** [maciejkula/spotlight#185]
Hello! Thank you for your great work!
In your realisation of bpr loss you use the following formula:

`    loss = (1.0 - torch.sigmoid(positive_predictions -
                                negative_predictions))
`.

In the original publication [(bpr paper)](https://arxiv.org/pdf/1205.2618.pdf) the authors propose
`loss = log(sigmoid(positive - negative)) + regularisation`. 
Is it okay?
